hazelcast/src/test/java/com/hazelcast/internal/cluster/impl/MembershipFailureTest.java

```diff
@@ -351,6 +359,19 @@ public void slave_receives_member_list_from_non_master() {
         dropOperationsFrom(slave2, F_ID, singletonList(HEARTBEAT));
         dropOperationsFrom(slave3, F_ID, singletonList(HEARTBEAT));
 
+        // To make sure that there is not pending/remaining heartbeat operations left in slave2
+        // and slave3, we're sending new urgent operations to these members that will be processed
+        // after those remaining heartbeats and then waiting for them to be processed.
+        OperationService masterOperationService = getOperationService(master);
+        masterOperationService.send(new UrgentOperationAwaitOn(), getNode(slave2).address);
+        masterOperationService.send(new UrgentOperationAwaitOn(), getNode(slave3).address);
+        OperationService slave1OperationService = getOperationService(slave1);
+        slave1OperationService.send(new UrgentOperationAwaitOn(), getNode(slave2).address);
+        slave1OperationService.send(new UrgentOperationAwaitOn(), getNode(slave3).address);
+
+        latchSlave2.await();
+        latchSlave3.await();
```

Comment:
Can it be an alternative solution if we wait at least `MAX_NO_HEARTBEAT_SECONDS` before suspicions?

Reply:
If we do that, the members will automatically suspect each other before our suspect requests at the bottom executed. This will change the test.

Reply:
Even in current test, with or without your fix, we cannot guarantee suspicion before explicit suspect calls. There can be unexpected pauses which can cause early suspicions. 

Good part of your fix is, we don't need to wait `MAX_NO_HEARTBEAT_SECONDS` always and test will finish quicker, bad part is one more internal api usage and this increases complexity of test.

This trade-off can be seen ok i think.